testsuite/MDAnalysisTests/analysis/test_persistencelength.py

```diff
@@ -149,18 +149,29 @@ def test_sortbb(self, u):
 
         assert_equal(s_ag.ids, [0, 1, 4, 6, 8])
 
-    def test_not_fragment(self, u):
-        # two fragments don't work
-        bad_ag = u.residues[0].atoms[:2] + u.residues[1].atoms[:2]
-        with pytest.raises(ValueError):
-            polymer.sort_backbone(bad_ag)
-
     def test_branches(self, u):
         # includes side branches, can't sort
         bad_ag = u.atoms[:10]  # include -H etc
 
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError) as ex:
             polymer.sort_backbone(bad_ag)
+        assert "branches or isolated atoms" in str(ex.value)
+
+    def test_isolated(self, u):
+        u = mda.Universe.empty(4, trajectory=True)
+        bondlist = [(0, 1), (1, 2)]
+        u.add_TopologyAttr(Bonds(bondlist))
+        with pytest.raises(ValueError) as ex:
+            polymer.sort_backbone(u.atoms)
+        assert "branches or isolated atoms" in str(ex.value)
+
+    def test_missing_internal(self, u):
+        u = mda.Universe.empty(4, trajectory=True)
+        bondlist = [(0, 1), (2, 3)]
+        u.add_TopologyAttr(Bonds(bondlist))
+        with pytest.raises(ValueError) as ex:
+            polymer.sort_backbone(u.atoms)
+        assert "Backbone connectivity invalid" in str(ex.value)
```

Comment:
When writing new tests it might be preferable to use the built-in `match=...` argument of `pytest.raises()` to check the error message that way. It is also borderline if we should just parametrize this test with the one above it--perhaps they're different enough not to bother.